nav2_route/src/goal_intent_extractor.cpp

```diff
@@ -193,6 +253,90 @@ Route GoalIntentExtractor::pruneStartandGoal(
   return pruned_route;
 }
 
+unsigned int GoalIntentExtractor::associatePoseWithGraphNode(
+  std::vector<unsigned int> node_indices,
+  const geometry_msgs::msg::PoseStamped & pose)
+{
+  unsigned int s_mx, s_my, g_mx, g_my;
+  if (!costmap_sub_->getCostmap()->worldToMap(
+      pose.pose.position.x, pose.pose.position.y,
+      s_mx, s_my))
+  {
+    throw nav2_core::StartOutsideMapBounds(
+            "Start Coordinates of(" + std::to_string(pose.pose.position.x) + ", " +
+            std::to_string(pose.pose.position.y) + ") was outside bounds");
+  }
+
+  if (costmap_sub_->getCostmap()->getCost(s_mx, s_my) == nav2_costmap_2d::LETHAL_OBSTACLE) {
+    throw nav2_core::StartOccupied("Start was in lethal cost");
+  }
+
+  //TODO(jwallace42): double check the frames, probably need to move into the map frame...
+  std::vector<unsigned int> valid_node_indices;
+  std::vector<nav2_costmap_2d::MapLocation> goals;
+  for (const auto & node_index : node_indices) {
+    float goal_x = (*graph_)[node_index].coords.x;
+    float goal_y = (*graph_)[node_index].coords.y;
+    if (!costmap_sub_->getCostmap()->worldToMap(goal_x, goal_y, g_mx, g_my)) {
+      RCLCPP_WARN_STREAM(
+        logger_, "Goal coordinate of(" + std::to_string(goal_x) + ", " +
+        std::to_string(goal_y) + ") was outside bounds. Removing from goal list");
+      continue;
+    }
+
+    if (costmap_sub_->getCostmap()->getCost(g_mx, g_my) == nav2_costmap_2d::LETHAL_OBSTACLE) {
+      RCLCPP_WARN_STREAM(
+        logger_, "Goal corrdinate of(" + std::to_string(goal_x) + ", " +
+        std::to_string(goal_y) + ") was in lethal cost. Removing from goal list.");continue;
+    }
+    valid_node_indices.push_back(node_index);
+    goals.push_back({g_mx, g_my});
+  }
+
+  if (goals.empty()) {
+    throw nav2_core::PlannerException("All goals were invalid");
+  }
+
+  bfs_->clearGraph();
+  bfs_->setStart(s_mx, s_my);
+  bfs_->setGoals(goals);
+  if (bfs_->isNodeVisible()) {
+    // The visiblity check only validates the first node in goal array
+    return valid_node_indices.front();
+  }
+
+  unsigned int goal;
+  bfs_->search(goal);
+
+  return valid_node_indices[goal];
+}
+
+void GoalIntentExtractor::visualizeExpansions(
+  rclcpp::Publisher<nav_msgs::msg::OccupancyGrid>::SharedPtr occ_grid_pub)
+{
+  if (!enable_search_viz_) {
+    return;
+  }
+  nav_msgs::msg::OccupancyGrid occ_grid_msg;
+  unsigned int width = costmap_sub_.get()->getCostmap()->getSizeInCellsX();
```

Comment:
Why are you `.get()` this? you can just `->` its a shared pointer, you don't need its underlying memory, that's what the accessor does

Reply:
no idea haha, 

this was late one night :). I should stop programming after 11.

Reply:
hah No worries :-)